dart/dynamics/SingleDofJoint.cpp

```diff
@@ -474,6 +474,19 @@ double SingleDofJoint::getPositionUpperLimit(size_t _index) const
 }
 
 //==============================================================================
+bool SingleDofJoint::hasPositionLimit(size_t _index) const
+{
+  if (_index != 0)
+  {
+    SINGLEDOFJOINT_REPORT_OUT_OF_RANGE( hasPositionLimit, _index );
+    return true;
+  }
+
+  return !(std::isinf(mSingleDofP.mPositionLowerLimit)
+        && std::isinf(mSingleDofP.mPositionUpperLimit));
```

Comment:
How about this:
```cpp
return (std::isfinite(mSingleDofP.mPositionLowerLimit)
        || std::isfinite(mSingleDofP.mPositionUpperLimit))
```
Less use of `not` operator seems more clear to understand to me. `std::isfinite` is opposite function to `std::isinf` except for `NAN` value case.  Both of them return `false` for `NAN` but I think `NAN` value is meaningless anyways.